lbjava/src/main/java/edu/illinois/cs/cogcomp/lbjava/learn/StochasticGradientDescent.java

```diff
@@ -193,11 +215,19 @@ public void learn(int[] exampleFeatures, double[] exampleValues, int[] exampleLa
         assert exampleLabels.length == 1 : "Example must have a single label.";
 
         double labelValue = labelValues[0];
-        double multiplier =
-                learningRate
-                        * (labelValue - weightVector.dot(exampleFeatures, exampleValues) - bias);
-        weightVector.scaledAdd(exampleFeatures, exampleValues, multiplier);
-        bias += multiplier;
+        double wtx = weightVector.dot(exampleFeatures, exampleValues) + bias;
+
+        if (isLMS) {
+            double multiplier = learningRate * (labelValue - wtx);
+            weightVector.scaledAdd(exampleFeatures, exampleValues, multiplier);
+            bias += multiplier;
+        } else {
```

Comment:
That's better. 

Even better than that is: having a single SGD which works for both classification and regression, and the user has the option of setting either LMS or Hinge, depending on what they need (classification/regression/etc).